### ISSUE TITLE
PDF viewer

### ISSUE THREAD

**Issue by platipo** [silverwind/droppy#266]
It's possible to add a PDF viewer like [pdf.js](https://mozilla.github.io/pdf.js/)?

**Comment by silverwind** [comment]
Surely possible. I just don't see much use for it personally, as I prefer the vastly faster SumatraPDF desktop app.

**Comment by platipo** [comment]
Thanks for your always fast reply. :smile:  
I see your point, but I uploaded lots PDFs and it's a pain in the neck looking for a specific one or manage the flooded Download directory. It would be terrific if you manage to insert the PDF viewer.

**Comment by silverwind** [comment]
I'll keep it in mind for the future. If it's going to be pdf.js, it'll require some re-skin work, partly because:

> However, we do ask if you plan to embed the viewer in your own site, that it not just be an unmodified version. Please re-skin it or build upon it.

**Comment by platipo** [comment]
It could look like this!
![preview](https://cloud.githubusercontent.com/assets/22260641/18627193/dd505760-7e59-11e6-9f69-5e1bcacbd315.png)

**Comment by silverwind** [comment]
[Progress 🎉](https://github.com/silverwind/droppy/commit/1db8496cb2a34b60641dd71dbe429f80cf8411c1)

![](https://cloud.githubusercontent.com/assets/115237/19947746/d50c8192-a149-11e6-9132-a6ef311ccc47.png)

I want to integrate this in media galleries and I have a few more optimizations in mind, so it'll require some rewrites and the PDF UI needs to be coded up from scratch to fit with the current gallery. Also, that 2.1 MB script should be properly cached which it isn't yet.

So still some ways to go.

**Comment by platipo** [comment]
Any update on this feature?

**Comment by silverwind** [comment]
No, I kind of forgot about it, but I might pick it up again later.

**Comment by silverwind** [comment]
And it's done:

<img width="1270" src="https://user-images.githubusercontent.com/115237/36177038-60e66024-110c-11e8-95f4-af86c94e97e4.png">
